druntime/src/core/internal/newaa.d

```diff
@@ -140,5 +151,18 @@ AAShell makeAA(K, V)(V[K] src)
     } ();
     // return the new implementation
     return AAShell(new Impl(buckets, cast(uint)srclen, 0, typeid(E), firstUsed,
-            K.sizeof, V.sizeof, E.value.offsetof, flags));
+            K.sizeof, V.sizeof, E.value.offsetof, flags, hashFn));
+}
+
+version(unittest):
+struct Foo {
+    ubyte x;
+    double d;
+}
+int[Foo] utaa = [Foo(1, 2.0) : 5];
+unittest {
```

Comment:
```suggestion
unittest
{
    static struct Foo
    {
        ubyte x;
        double d;
    }
    static int[Foo] utaa = [Foo(1, 2.0) : 5];
```

Reply:
https://issues.dlang.org/show_bug.cgi?id=24209

Reply:
I'll look into it

Reply:
https://github.com/dlang/dmd/pull/15774